TUM4Health/data-management-service
# [feature/zhs-scraping] Scraping of the ZHS web page

# Scraping of the ZHS web page

## :recycle: Current situation & Problem
Currently, there is no convenient way to access data of the [ZHS website](https://www.zhs-muenchen.de/zhs/startseite/)

## :bulb: Proposed solution
Build a web scraper with [Cheerio](https://cheerio.js.org/) that scrapes the website data, formats it and puts it into an ordered database structure model

## :gear: Release Notes 
- Add ZHS web scraping

## :heavy_plus_sign: Additional Information
As the ZHS website is quite unreliably, the PR includes multiple fail-safes like automatic request retry and limitation of connections to the website

### Related PRs
None

### Testing
- Insert a scraper object into the Strapi database via the Content Manager (`next_execution_at` and `currentlyRunning` are set via Strapi):
<img width="892" alt="image" src="https://user-images.githubusercontent.com/25406915/172049831-fb93b3b1-eeea-46c7-a1dd-1eef40f3c7d2.png">
- Take a look at the logs to see if new entries are fetched and created/updated
- Check the Content Manager of Strapi to verify that the data is fetched and put into the database correctly

### Reviewer Nudging
`src/scrapers/zhsofferings.js` should be a good starting point. Take a look at how the whole scraping process is done as well as how the scraper is triggered
